File: src/services/Agenda.py
from models.Contato import Contato

class Agenda:
    def __init__(self):
        self.contatos = []

    def adicionar_contatos(self, contato):
        self.contatos.append(contato)

    def remover_contato(self, nome):
        contato_remover = None
        for contato in self.contatos:
            if contato.nome.lower() == nome.lower():
                contato_remover = contato
                break

        if contato_remover:
            self.contatos.remove(contato_remover)
            print(f"Contato removido: {nome}")
        else:
            print("Contato não encontrado.")

    def buscar_contato(self, termo_busca):
        for contato in self.contatos:
            if (contato.nome.lower() == termo_busca.lower() or
                contato.telefone == termo_busca or 
                contato.email.lower() == termo_busca.lower()):
                return contato
        return None
    
    def listar_contatos(self):
        if not self.contatos:
            print("A agenda está vazia.")
        else:
            print("Contatos na agenda:")
            for contato in self.contatos:
                contato.exibir_detalhes()
                print("---------------")

File: src/services/Agenda.java
package services;

import models.Contact;

import java.util.ArrayList;

public class Agenda {
    private ArrayList<Contact> contacts;

    public Agenda() {
        contacts = new ArrayList<>();
    }

    public void addContacts(Contact contact){
        contacts.add(contact);
        System.out.println(contact.getName() + " Foi adicionado a sua lista de contatos");
    }

    public Contact removeContacts(String name) {
        Contact contactRemove = null;

        for (Contact contact : contacts) {
            if (contact.getName().equalsIgnoreCase(name)) {
                contactRemove = contact;
                break;
            }
        }

        if (contactRemove != null) {
            contacts.remove(contactRemove);
            System.out.println(name + " Foi removido da sua lista de contatos.");
        } else {
            System.out.println("Contato não encontrado.");
        }
        return contactRemove;
    }

    public Contact searchContacts(String searchTerm) {
        for (Contact contact : contacts) {
            if (contact.getName().equalsIgnoreCase(searchTerm) ||
                contact.getTelephone().equals(searchTerm) ||
                contact.getEmail().equalsIgnoreCase(searchTerm)) {
                System.out.println("---------------");
                System.out.println("Contato Encontrado:");
                return contact;
            }
        }
        return null;
    }

    public void listContacts() {
        if (contacts.isEmpty()) {
            System.out.println("Agenda está vazia.");
        } else {
            System.out.println("Contatos na agenda:");

            for (Contact contact : contacts) {
                contact.showDetails();
                System.out.println("---------------");
            }
        }
    }
}
